composeApp/src/commonTest/kotlin/io/github/couchtracker/db/user/model/icon/DbIconTest.kt

```diff
@@ -0,0 +1,58 @@
+package io.github.couchtracker.db.user.model.icon
+
+import io.github.couchtracker.db.user.CouchTrackerUri
+import io.github.couchtracker.utils.parseUri
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.datatest.withData
+import io.kotest.matchers.shouldBe
+
+class DbIconTest : FunSpec(
+    {
+        context("parseUri()") {
+            context("default icon") {
+                context("valid URIs return the correct icon") {
+                    withData(
+                        "couch-tracker://icon/default/home" to DbDefaultIcon.HOME.toDbIcon(),
+                        "couch-tracker://icon/default/cinema" to DbDefaultIcon.CINEMA.toDbIcon(),
+                        "couch-tracker://icon/default/office-building" to DbDefaultIcon.OFFICE_BUILDING.toDbIcon(),
+                    ) { (uriString, expectedIcon) ->
+                        val uri = CouchTrackerUri(parseUri(uriString))
+                        DbIcon.fromUri(uri) shouldBe expectedIcon
+                    }
+                }
+                context("invalid default icon returns unknown icon") {
+                    withData(
+                        "couch-tracker://icon/default/i-dont-exist",
+                        "couch-tracker://icon/default/aaaaaaaaaaaaaaaaa",
+                    ) { uriString ->
+                        val uri = CouchTrackerUri(parseUri(uriString))
+                        DbIcon.fromUri(uri) shouldBe DbIcon.UnknownDefault(uri)
+                    }
+                }
+                context("invalid default URIs fail") {
+                    withData(
+                        "couch-tracker://icon/default",
+                        "couch-tracker://icon/default/",
+                        "couch-tracker://icon/default/%20",
+                    ) { uriString ->
+                        val uri = CouchTrackerUri(parseUri(uriString))
+                        shouldThrow<IllegalArgumentException> {
```

Comment:
Question: is `IllegalArgumentException` the right exception for ColumnAdapter? E.g. are other implementations throwing `ParseException` or something?

Reply:
I don't see any specific exception required in the documentation, I think any will do. We are already using `IllegalArgumentException` for the partial date time adapter.

For instance, the default `EnumColumnAdapter` uses `.first { }`, which throws `NoSuchElementException`: https://github.com/cashapp/sqldelight/blob/598638f0adb449ca1f3c4a30fafd20210d59c7d8/runtime/src/commonMain/kotlin/app/cash/sqldelight/EnumColumnAdapter.kt#L22